compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -30,6 +30,116 @@
 #include "il/Node_inlines.hpp"
 #include "infra/Bit.hpp"
 
+typedef enum RegisterStatus {
+   OK_FOR_REUSE,
+   DO_NOT_TOUCH
+} RegisterStatus;
+
+static inline RegisterStatus
+evaluateRegister(TR::Register **childReg, TR::Node *child, TR::Node *parent, TR::CodeGenerator *cg)
+   {
+   int64_t value;
+   RegisterStatus status;
+
+	if (child->getOpCode().isLoadConst() && child->getRegister() == NULL)
+   {
+      value = child->getLongInt();
+      if (constantIsUnsignedImm12(value))
+      {
+         //TODO: Cover special cases for immediate loads
+      }
+      else
+      {
+         *childReg = cg->allocateRegister();
+         loadConstant64(cg, parent, value, *childReg);
+         status = OK_FOR_REUSE;
+      }
+	}
+	else
+   {
+      *childReg = cg->evaluate(child);
+      status = DO_NOT_TOUCH;
+	}
+   return status;
+   }
+
+static inline TR::Register *
+evaluateBooleanTypeInteger(TR::Node *node, TR::InstOpCode::Mnemonic regOp, TR::CodeGenerator *cg, uint8_t sizeOfOperation)
+   {
+   TR::Register *srcReg[3];
+   RegisterStatus srcStatus[3];
+   TR::Node *children[3];
```

Comment:
Do you need the `children` array?  It seems the children always map to a particular child on the Node and there's no need to keep them in a separate array.

Reply:
I believe we included an array to reduce the amount of code (some of the repeated if statements with the children can be converted into for loops, and we did this in a slightly newer version we have). If you would prefer to keep arrays out of it, we can change it. :)

Reply:
In this implementation, the elements of `children` always map to the corresponding ordinal child position on the node, hence it is not needed.  If there is newer code that hasn't been pushed then I can reserve this comment until I see that code.